Build for glibc 2.17 and libcap 2.75

ehfd opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

Closes #37
Closes #39
Closes #41

<!--
Please add any other relevant info below:
-->

ehfd committed:
Build for Glibc 2.17 and libcap 2.71

ehfd commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

ehfd committed:
Add maintainer

Commit:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.46.1, and conda-forge-pinning 2025.03.08.23.19.55

ehfd commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.46.1, and conda-forge-pinning 2025.03.08.23.19.55

ehfd committed:
Fix build

ehfd commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.46.1, and conda-forge-pinning 2025.03.09.19.15.19

ehfd committed:
Edit dependencies

ehfd commented:
@conda-forge-admin, please rerender

ehfd committed:
Add dependencies

ehfd commented:
@conda-forge-admin, please rerender

ehfd committed:
Fix packages

ehfd commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.46.1, and conda-forge-pinning 2025.03.09.19.15.19

ehfd commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13752167763](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13752167763). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13752174405](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13752174405). Examine the logs at this URL for more detail.</sub>

ehfd committed:
Modify dependencies

ehfd commented:
@ryanvolz

```
#16 200.5 Encountered problems while solving:
#16 200.5   - package libudev1-257.4-h9a4d06a_0 requires libcap >=2.71,<2.72.0a0, but none of the providers can be installed
#16 200.5 
#16 200.5 Could not solve for environment specs
#16 200.5 The following packages are incompatible
#16 200.5 ├─ libudev1 >=257.4  is installable with the potential options
#16 200.5 │  ├─ libudev1 257.4 would require
#16 200.5 │  │  └─ libcap >=2.71,<2.72.0a0 , which can be installed;
#16 200.5 │  └─ libudev1 257.4 would require
#16 200.5 │     └─ libcap >=2.75,<2.76.0a0 , which can be installed;
#16 200.5 ├─ pulseaudio is installable with the potential options
#16 200.5 │  ├─ pulseaudio 17.0 would require
#16 200.5 │  │  └─ pulseaudio-daemon 17.0 haa314ec_0, which requires
#16 200.5 │  │     └─ libcap >=2.69,<2.70.0a0 , which conflicts with any installable versions previously reported;
#16 200.5 │  └─ pulseaudio [13.0|14.0|16.1], which can be installed;
#16 200.5 └─ pulseaudio-client >=17.0,<17.1.0a0  is not installable because it requires
#16 200.5    └─ pulseaudio 17.0 *_0, which conflicts with any installable versions previously reported.
```

Any help possible? This is still all over the place...

ehfd committed:
Small edits

ehfd committed:
Fix iconv

ehfd committed:
Fix line endings

ehfd commented:
- libiconv

ehfd commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.47.0, and conda-forge-pinning 2025.03.19.14.45.19

ehfd committed:
Fix patch

ryanvolz committed:
Remove dependencies that don't get linked

ryanvolz committed:
Add libltdl dep to daemon, needed after libtool package split outputs

ryanvolz committed:
Remove libiconv dep from daemon since it doesn't link it

ryanvolz committed:
Ignore run exports from glib for daemon, since it doesn't get linked

I think it is still needed as a dependency so that meson finds it and
enables features that we want enabled.

ryanvolz committed:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.47.0, and conda-forge-pinning 2025.03.19.14.45.19

ryanvolz commented:
I think I understand what was happening with libiconv: pulseaudio was using meson to check whether the compiler provides iconv functions (it does) and does not check for the libiconv library, but then libiconv was in the host environment (from glib) so its headers were being used instead, and the libiconv headers use different function names than the compiler built-in iconv and those were not being found since libiconv was not being linked. Your patch to skip the detection and just find libiconv and link to it is the way to go given that we have to have it in the environment.

Now, I'm not sure why the test is segfaulting on linux-64, but I can see from the other builds that there's a dependency missing (libltdl) and others that are not needed, so I'm going to try cleaning those up to see if it works.

ryanvolz commented:
Progress, but still the `cpu-volume-test` is segfaulting.

ehfd commented:
https://github.com/conda-forge/pulseaudio-feedstock/pull/37#issuecomment-2543026536

@ryanvolz 

Hi, Ryan. Please check the details I found above regarding this.
The verdict is that the segfault is because of "unrealistic" test settings, and while the fixes for gstreamer-orc are in conda-forge (v0.4.40 and later), the test fix for components within PulseAudio has not arrived yet.
So it seems that we are safe to skip this test, only for v17.0.

Moreover, it seems that fftw3 is still a dependency in the client, so I brought it back.

ehfd committed:
Fix build pipeline

ehfd commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13966434513](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13966434513). Examine the logs at this URL for more detail.</sub>

ehfd committed:
Fix overlinking

ehfd commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.1.2, conda-smithy 3.47.0, and conda-forge-pinning 2025.03.19.20.07.31

ehfd committed:
Skip test for v17.0

ehfd committed:
Skip issued tests

ehfd commented:
@conda-forge-admin, please rerender

ehfd committed:
Fix X11 build

ehfd commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13966826355](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13966826355). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13966870115](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13966870115). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13966874612](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13966874612). Examine the logs at this URL for more detail.</sub>

ehfd committed:
Fix line endings

ehfd commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13967099012](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13967099012). Examine the logs at this URL for more detail.</sub>

ehfd committed:
Fix overlinking

ehfd commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13967275004](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13967275004). Examine the logs at this URL for more detail.</sub>

ehfd committed:
Add libsamplerate

ehfd commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13967507244](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13967507244). Examine the logs at this URL for more detail.</sub>

ehfd committed:
Delete nonexistent dependency

ehfd commented:
@conda-forge-admin, please rerender

ehfd commented:
All done, with all overdepending packages removed. @ryanvolz 

Refer to https://github.com/conda-forge/pulseaudio-feedstock/pull/40#issuecomment-2739132060 in case you missed.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13967627698](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13967627698). Examine the logs at this URL for more detail.</sub>

ryanvolz committed:
Upstream patch to fix cpu-volume-test

ryanvolz commented:
I'm inclined to backport the upstream patch that fixes cpu-volume-test rather than apply a patch to skip the test, because with the latter there's a strong likelihood that we'll forget to re-enable the test with the next release and with the former we'll be forced to drop the patch when it fails to apply for the next release.

Otherwise, this looks good.

ryanvolz commented:
Thanks for the PR, and welcome to the maintainer team!

ehfd commented:
@ryanvolz @hmaarrfk One more remark. It seems that v17.0 is a development release and the stable releases are v16.x (and later on v18.x). It might be needed to back off to v16.2.
